be/src/exprs/agg/percentile_cont.h

```diff
@@ -40,11 +40,120 @@ template <LogicalType LT, typename = guard::Guard>
+    CppType minV = RunTimeTypeLimits<LT>::min_value();
+    CppType maxV = RunTimeTypeLimits<LT>::max_value();
+
+    b.resize(k + 1);
+    ls.resize(k);
+    for (int i = 0; i < k; ++i) {
+        if constexpr (reverse) {
+            mp[i] = grid[i].size() - 2;
+        } else {
+            mp[i] = 1;
+        }
+    }
+    for (int i = 0; i < k; ++i) {
+        b[i] = grid[i][mp[i]];
+        if constexpr (reverse) {
+            mp[i]--;
+        } else {
+            mp[i]++;
+        }
+    }
+    b[k] = reverse ? maxV : minV;
+    for (int i = 0; i < k; ++i) {
+        ls[i] = k;
+    }
+
+    for (int i = k - 1; i >= 0; --i) {
+        int q = i;
+        int t = (q + k) / 2;
+        while (t > 0) {
+            if constexpr (reverse) {
+                if (b[q] < b[ls[t]]) {
+                    std::swap(q, ls[t]);
+                }
+            } else if (b[q] > b[ls[t]]) {
+                std::swap(q, ls[t]);
+            }
+            t = t / 2;
+        }
+        ls[0] = q;
+    }
+
+    CppType tp = reverse ? minV : maxV;
+    size_t cnt = 0;
+
+    while (b[ls[0]] != tp) {
+        int q = ls[0];
+        if (UNLIKELY(cnt >= goal)) {
+            if (cnt == goal) {
+                if constexpr (reverse)
+                    seniorElm = b[q];
+                else
+                    juniorElm = b[q];
+            }
+            if (cnt == goal + 1) {
+                if constexpr (reverse)
+                    juniorElm = b[q];
+                else
+                    seniorElm = b[q];
+                break;
+            }
+        }
+        cnt++;
+        b[q] = grid[q][mp[q]];
+
+        if constexpr (reverse) {
+            mp[q]--;
+        } else {
+            mp[q]++;
+        }
+        int t = (q + k) / 2;
+        while (t > 0) {
+            if constexpr (reverse) {
+                if (b[q] < b[ls[t]]) {
+                    std::swap(q, ls[t]);
+                }
+            } else if (b[q] > b[ls[t]]) {
+                std::swap(q, ls[t]);
+            }
+            t = t / 2;
+        }
+        ls[0] = q;
+    }
+}
+
+template <LogicalType LT, typename CppType, typename ResultType>
+ResultType calculateResult(CppType juniorElm, CppType seniorElm, double u, size_t index) {
+    [[maybe_unused]] ResultType result;
+    if constexpr (lt_is_datetime<LT>) {
+        result.from_unix_second(juniorElm.to_unix_second() +
+                                (u - (float)index) * (seniorElm.to_unix_second() - juniorElm.to_unix_second()));
+    } else if constexpr (lt_is_date<LT>) {
+        result._julian = juniorElm._julian + (u - (double)index) * (seniorElm._julian - juniorElm._julian);
+    } else if constexpr (lt_is_arithmetic<LT>) {
+        result = juniorElm + (u - (double)index) * (seniorElm - juniorElm);
+    } else {
+        // won't go there if percentile_cont is registered correctly
+        throw std::runtime_error("Invalid LogicalTypes for percentile_cont function");
```

Comment:
Please report the specific exceptional type name

Reply:
Sure, already added